### PR TITLE
add a hashivault_token_role module

### PR DESCRIPTION
**Pull request by Agalin**
This module facilitates creating, updating and deleting roles in a
specified token auth engine.

This module supports check and diff mode to allow verifying potential
changes before actually applying them.

The code in this commit has been formatted with `black`.

### CODE REVIEW

**Comment by TerryHowe** [comment]
Could you add some basic test if they pass
``% ls test*role*
test_approle.yml		test_approle_mount_point.yml	test_aws_auth_role.yml		test_azure_role.yml		test_db_role.yml
test_approle_check_mode.yml	test_approle_old.yml		test_azure_auth_role.yml	test_consul_role.yml		test_oidc_auth_role.yml
``

Rebase should solve the doc test failure, but no matter

**Comment by Agalin** [comment]
I'll go back to it on Wednesday. BTW please don't be surprised - there will be ~10 similar in size pull requests coming - we're upstreaming our changes.

File: ansible/modules/hashivault/hashivault_token_role.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    # make the changes!
+    if changed and state == "present":
+        if not module.check_mode:
+            # Before posting the desired state to the vault api we need to
+            # normalize some keys. This is a quirk of the vault api that it
+            # expects a different data format in the PUT/POST endpoint than
+            # it returns in the GET endpoint.
+            extra_params = {}
+
+            doc = yaml.safe_load(DOCUMENTATION)
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_token_role.py:274]: clever and scary at the same time, but there are tests here!

**Comment by TerryHowe** [comment]
Thanks!

**Pull request merged**